### ISSUE TITLE
`Style/HashSlice` gives false positive for array of strings

### ISSUE THREAD

**Issue by MatheusRich** [rubocop/rubocop#14511]
## Expected behavior

No offense should be reported.

## Actual behavior

Rubocop reports an offense for `Style/HashSlice` even when using an array. It seems to only happens if the compared value on `==` is a string. Worse, the cop suggests a autocorrection of `slice("1")` which plainly doesn't work on arrays (as it expects an integer arg).

## Steps to reproduce the problem

```rb
array = [["1", 1], ["1", 2], ["2", 1], ["2", 2]]
array.filter { |f, _| f == "1" }
#
```

## RuboCop version

```
1.80.2 (using Parser 3.3.9.0, rubocop-ast 1.46.0, analyzing as Ruby 3.2, running on ruby 3.4.2) [x86_64-darwin24]
```

**Comment by koic** [comment]
As already explained in the documentation, RuboCop cannot know through static analysis whether the receiver is a Hash object, which is why it is already marked as unsafe:
https://docs.rubocop.org/rubocop/1.80/cops_style.html#safety-stylehashslice
